martian/src/lib.rs

```diff
@@ -257,12 +259,14 @@ fn martian_entry_point<S: std::hash::BuildHasher>(
         },
     ));
 
-    let result = if md.stage_type == "split" {
-        stage.split(&mut md)
-    } else if md.stage_type == "main" {
-        stage.main(&mut md)
-    } else if md.stage_type == "join" {
-        stage.join(&mut md)
+    let md = Rc::new(RefCell::new(md));
+
+    let result = if md.borrow().stage_type == "split" {
+        stage.split(md.clone())
+    } else if md.borrow().stage_type == "main" {
+        stage.main(md.clone())
+    } else if md.borrow().stage_type == "join" {
```

Comment:
Can `match md.borrow().stage_type` be used here?

Reply:
Yes, will refactor

Reply:
Ah, actually, no, because of how the borrows are scoped.  If we match, then the borrow into the metadata is maintained over the entire scope of the match, which causes a reborrow error inside the stage code itself.